benchmarks/core/assumption.py

```diff
@@ -0,0 +1,419 @@
+# -*- coding: utf-8 -*-
+
+from sympy import pi, oo
+from sympy.core import Symbol
+from sympy.core import Symbol, S, Rational, Integer
+
+
+class Time_assumption:
+
+    def setup(self):
+        self.x = Symbol('x', real=True, integer=True)
+        self.z_0 = Integer(0)
+        self.z_1 = Integer(1)
+        self.z__1 = Integer(-1)
+        self.oo = S.Infinity
+        self.mm = S.NegativeInfinity
+        self.zoo = S.ComplexInfinity
+        self.nan = S.NaN
+        self.r_pos = Rational(3, 4)
+        self.r_neg = Rational(-3, 4)
+        self.pi = S.Pi
+        self.ex = S.Exp1
+        self.I = S.ImaginaryUnit
+        self.a_1 = Symbol('a', real=False)
+        self.a_2 = Symbol('a', extended_real=False)
+        self.a_3 = Symbol('a', imaginary=True)
+        self.x_0 = Symbol('x', zero=True)
+        self.x_1 = Symbol('x', positive=True)
+        self.x_2 = Symbol('x', nonpositive=True)
+        self.x_3 = Symbol('x', positive=False)
+
+    def time_symbol_unset(self):
+
+        self.x.is_real is True
+        self.x.is_integer is True
+        self.x.is_imaginary is False
+        self.x.is_noninteger is False
+        self.x.is_number is False
+
+
+    def time_zero(self):
+
+	    self.z_0.is_commutative is True
+	    self.z_0.is_integer is True
+	    self.z_0.is_rational is True
+	    self.z_0.is_algebraic is True
```

Comment:
I don't think it's necessary to test all of these. Benchmarks should be as narrow as possible (they should time precisely one thing). Also After checking `z_0.is_integer` the results of the other operations will be cached for any Basic instance anyway e.g.:
```julia
In [1]: a = Add(2, 2, evaluate=False)                                                                                                          

In [2]: a._assumptions                                                                                                                         
Out[2]: {}

In [3]: a.is_integer                                                                                                                           
Out[3]: True

In [4]: a._assumptions                                                                                                                         
Out[4]: 
{'integer': True,
 'algebraic': True,
 'rational': True,
 'hermitian': True,
 'commutative': True,
 'irrational': False,
 'complex': True,
 'infinite': False,
 'imaginary': False,
 'transcendental': False,
 'finite': True,
 'extended_real': True,
 'noninteger': False,
 'real': True}
```